### DISCUSSION TITLE
Vector stops processing Exec streaming logs after ~12hrs from python script command

### DISCUSSION THREAD

**Discussion by tegantaylor** [vectordotdev/vector#16109] [General]
After about 12 hours, vector just stops processing logs coming in from a python script - running "vector top" shows other log sources are still processing, but the counters for the exec source logs just stop and don't increase. 
If I look at the running processes on the box, I can see there is still a vector pid for the /usr/bin/python3 /etc/vector/scripts/log_script.py.
If i run the python script manually and output to a text file on the system, I can see new logs coming through and writing to the log file.

Here is the vector config:
---
sources:
  s_log_test:
    type: exec
    mode: streaming
    command: ["/etc/vector/scripts/log_script.py"]
    include_stderr: false
    streaming:
      respawn_on_exit: true
transforms:
  t_log_test:
    type: remap
    inputs: [s_log_test]
    source: |
      del(.source_type)
      . = parse_json!(.message)
      .source_type = "test"
      .host = "hostname.x.x.x"

**Comment by davidhuie-dd** [comment]
Is there complete confidence that the script is still running? You could `strace` the python script while it's in that apparently locked state to see if there's any activity. Another thing you could try is duplicating the logs to disk, just to confirm that Vector's the problem.
